cmd/login/cmd.go

```diff
@@ -149,14 +161,40 @@ func run(cmd *cobra.Command, argv []string) {
 	}
 
 	if args.useAuthCode {
-		fmt.Println("You will now be redirected to Red Hat SSO login")
+		r.Reporter.Infof("You will now be redirected to Red Hat SSO login")
+		// Short wait for a less jarring experience
+		time.Sleep(2 * time.Second)
 		token, err := authentication.InitiateAuthCode(oauthClientId)
 		if err != nil {
-			r.Reporter.Errorf("An error occurred while retrieving the token : %v", err)
+			r.Reporter.Errorf("An error occurred while retrieving the token: %v", err)
 			os.Exit(1)
 		}
 		args.token = token
-		fmt.Println("Token received successfully")
+		args.clientID = oauthClientId
+		r.Reporter.Infof("Token received successfully")
+	}
+
+	if args.useDeviceCode {
+		deviceAuthConfig := &authentication.DeviceAuthConfig{
+			ClientID: oauthClientId,
+		}
+		_, err := deviceAuthConfig.InitiateDeviceAuth(ctx)
+		if err != nil || deviceAuthConfig == nil {
+			r.Reporter.Errorf("An error occurred while initiating device auth: %v", err)
```

Comment:
this will lead to confusing message when err==nil

Reply:
Good catch, removed the deviceAuthConfig condition. This will never be true.